### PR TITLE
Package: add new package py-fitter

### PR DESCRIPTION
**Pull request by carsonwoods**
Adds py-fitter as a new package. It is a python package which makes identifying statistical distributions for datasets faster. It is a fairly bare-bones package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-fitter/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    """fitter package provides a simple class to identify the distribution
+    from which a data samples is generated from. It uses 80 distributions
+    from Scipy and allows you to plot the results to check what is the
+    most probable distribution and the best parameters."""
+
+    homepage = "https://github.com/cokelaer/fitter"
+    pypi = "fitter/fitter-1.5.1.tar.gz"
+
+    # notify when the package is updated.
+    # maintainers = ["carsonwoods"]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-fitter/package.py]: Did you intend to remove the comment from this line so Spack can process the `maintainers` list?

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256.  I do have a question about the maintainers property.  Also recommend removing the comment at line 17.

File: var/spack/repos/builtin/packages/py-fitter/package.py (hunk 1 above)
Comment (carsonwoods) [reply var/spack/repos/builtin/packages/py-fitter/package.py]: Oh yes, that was an oversight. I've gone ahead and changed it.

**Review by tldahlgren — DISMISSED** [review]
LGTM

**Comment by tldahlgren** [comment]
Deferring merge decision to @adamjstewart 's expertise.

File: var/spack/repos/builtin/packages/py-fitter/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    maintainers = ["carsonwoods"]
+
+    version("1.5.1", sha256="893b35ad0a84c3b96b63ec203a6a79effdba98777aed966ae61709f5e1e8cf99")
+
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-numpy")
+    depends_on("py-scipy")
+    depends_on("py-tqdm")
+    depends_on("py-joblib")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fitter/package.py]: Missing several required deps:
```
click
joblib
matplotlib
numpy
pandas
scipy>=0.18
tqdm
```
Also, these should all be `type=('build', 'run')`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**